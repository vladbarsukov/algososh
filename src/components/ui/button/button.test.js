import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';
import {Button} from './button';


describe('Test ButtonComponent', () => {

  it("should render and Match Snapshot with text", () => {
   const props = {
      text: "button"
    }
    const ButtonComponent = renderer.create(<Button {...props}/>).toJSON();
    render(<Button {...props}/>)
    expect(screen.getByRole('button')).toHaveTextContent('butto');
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("should render and Match Snapshot without text", () => {
    const ButtonComponent = renderer.create(<Button/>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });
  it("should render and Match Snapshot with disabled button", () => {
    const props = {
      disabled: true
    }
    render(<Button {...props}/>)
    expect(screen.getByRole('button')).toBeDisabled()
    const ButtonComponent = renderer.create(<Button {...props}/>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("should render and match snapshot of button with load indication", () => {
    const props = {
      isLoader: true
    };
    render(<Button {...props} />);
    const body = document.body;
    expect(body).toContainElement(
      screen.getByAltText('Загрузка.')
    );
    const buttonComponent = renderer.create(<Button {...props} />).toJSON();
    expect(buttonComponent).toMatchSnapshot();
  });

  it("should click on button work correctly", () => {
    const onClickMock = jest.fn()
    const props = {
      text: "button",
      onClick: onClickMock,
    }
    render(<Button {...props}/>)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(onClickMock).toHaveBeenCalled()
  });
});