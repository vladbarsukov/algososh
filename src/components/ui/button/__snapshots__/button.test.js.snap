// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component should render and Match Snapshot button with indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Test Button component should render and Match Snapshot with disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test Button component should render and Match Snapshot with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    button
  </p>
</button>
`;

exports[`Test Button component should render and Match Snapshot without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test Button component should render and match snapshot of button with indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Test Button component should render and match snapshot of button with load indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Test Button component should render and match snapshot of button with load indication 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Test ButtonComponent should render and Match Snapshot with disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test ButtonComponent should render and Match Snapshot with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    button
  </p>
</button>
`;

exports[`Test ButtonComponent should render and Match Snapshot without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test ButtonComponent should render and match snapshot of button with load indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
